src/transformers/generation/configuration_utils.py

```diff
@@ -0,0 +1,545 @@
+                A dictionary of proxy servers to use by protocol or endpoint, e.g., `{'http': 'foo.bar:3128',
+                'http://hostname': 'foo.bar:4012'}.` The proxies are used on each request.
+            use_auth_token (`str` or `bool`, *optional*):
+                The token to use as HTTP bearer authorization for remote files. If `True`, or not specified, will use
+                the token generated when running `huggingface-cli login` (stored in `~/.huggingface`).
+            revision (`str`, *optional*, defaults to `"main"`):
+                The specific model version to use. It can be a branch name, a tag name, or a commit id, since we use a
+                git-based system for storing models and other artifacts on huggingface.co, so `revision` can be any
+                identifier allowed by git.
+
+                <Tip>
+
+                To test a pull request you made on the Hub, you can pass `revision="refs/pr/<pr_number>".
+
+                </Tip>
+
+            return_unused_kwargs (`bool`, *optional*, defaults to `False`):
+                If `False`, then this function returns just the final configuration object.
+
+                If `True`, then this functions returns a `Tuple(config, unused_kwargs)` where *unused_kwargs* is a
+                dictionary consisting of the key/value pairs whose keys are not configuration attributes: i.e., the
+                part of `kwargs` which has not been used to update `config` and is otherwise ignored.
+            subfolder (`str`, *optional*, defaults to `""`):
+                In case the relevant files are located inside a subfolder of the model repo on huggingface.co, you can
+                specify the folder name here.
+            kwargs (`Dict[str, Any]`, *optional*):
+                The values in kwargs of any keys which are configuration attributes will be used to override the loaded
+                values. Behavior concerning key/value pairs whose keys are *not* configuration attributes is controlled
+                by the `return_unused_kwargs` keyword parameter.
+
+        Returns:
+            [`GenerationConfig`]: The configuration object instantiated from this pretrained model.
+
+        Examples:
+
+        ```python
+        generation_config = GenerationConfig.from_pretrained(
+            "gpt2"
+        )  # Download configuration from huggingface.co and cache.
+        generation_config = GenerationConfig.from_pretrained(
+            "./test/saved_model/"
+        )  # E.g. config was saved using *save_pretrained('./test/saved_model/')*
+        generation_config = GenerationConfig.from_pretrained("./test/saved_model/", "my_configuration.json")
+        config = GenerationConfig.from_pretrained("gpt2", top_k=1)
+        assert config.top_k == 1
+        config, unused_kwargs = GenerationConfig.from_pretrained("gpt2", top_k=1, foo=False, return_unused_kwargs=True)
+        assert config.top_k == 1
+        assert unused_kwargs == {"foo": False}
```

Comment:
Instead of asserts, use the doctest syntax to show the results.

Edit: Saw the below. Let's add the config file in GPT-2 then!